tests/integration/test_kms.py

```diff
@@ -39,22 +55,58 @@ def test_create_key(self, kms_client, sts_client):
         assert f":{region}:" in response["Arn"]
         assert f":{account_id}:" in response["Arn"]
 
+    @pytest.mark.aws_validated
+    def test_schedule_and_cancel_key_deletion(self, kms_client, kms_create_key):
+        key_id = kms_create_key()["KeyId"]
+        kms_client.schedule_key_deletion(KeyId=key_id)
+        result = kms_client.describe_key(KeyId=key_id)
+        assert result["KeyMetadata"]["Enabled"] is False
+        assert result["KeyMetadata"]["KeyState"] == "PendingDeletion"
+        assert result["KeyMetadata"]["DeletionDate"]
+
+        kms_client.cancel_key_deletion(KeyId=key_id)
+        result = kms_client.describe_key(KeyId=key_id)
+        assert result["KeyMetadata"]["Enabled"] is False
+        assert result["KeyMetadata"]["KeyState"] == "Disabled"
+        assert not result["KeyMetadata"].get("DeletionDate")
+
+    @pytest.mark.aws_validated
+    def test_disable_and_enable_key(self, kms_client, kms_create_key):
+        key_id = kms_create_key()["KeyId"]
+        result = kms_client.describe_key(KeyId=key_id)
+        assert result["KeyMetadata"]["Enabled"] is True
+        assert result["KeyMetadata"]["KeyState"] == "Enabled"
+
+        kms_client.disable_key(KeyId=key_id)
+        result = kms_client.describe_key(KeyId=key_id)
+        assert result["KeyMetadata"]["Enabled"] is False
+        assert result["KeyMetadata"]["KeyState"] == "Disabled"
+
+        kms_client.enable_key(KeyId=key_id)
+        result = kms_client.describe_key(KeyId=key_id)
+        assert result["KeyMetadata"]["Enabled"] is True
+        assert result["KeyMetadata"]["KeyState"] == "Enabled"
+
+    # Not sure how useful this test is, as it just fails during key validation, before grant-specific logic kicks in.
     @pytest.mark.aws_validated
     def test_create_grant_with_invalid_key(self, kms_client, user_arn):
 
-        with pytest.raises(botocore.exceptions.ClientError):
+        with pytest.raises(botocore.exceptions.ClientError) as e:
             kms_client.create_grant(
                 KeyId="invalid",
                 GranteePrincipal=user_arn,
                 Operations=["Decrypt", "Encrypt"],
             )
+        assert e.match("NotFoundException")
```

Comment:
tiny nit: `assert` technically not needed - `match` already performs the assertion internally 👍

Reply:
Not my test. :-)
My responsibility, though. :-(

Done.